File: Blind 75 LeetCode/Chahat Rajput/Two Sum/TwoSum.cpp
class Solution {
public:
//Complexity O(n)
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> m;
       for(int i=0;i<nums.size();i++){
           //target ko array ke pehle element se minus kr do and vo element find kro jo target ka addition ho
           int element=target-nums[i];//9-2=7 find 7 from array
           if(m.find(element) != m.end()){
               return {m[element],i};
           }
            m[nums[i]]=i;
       }
        return{};
    }
};

File: Blind 75 LeetCode/Subhangi dhasmana/Two sum/two sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # for i in range(len(nums)):
        #     for j in range(i+1, len(nums)):
        #         if nums[i]+nums[j]==target:
        #             return [i, j]
        # return -1
        prevMap = {}
        for i in range(len(nums)):
            rem = target-nums[i]
            if rem in prevMap:
                return {prevMap[rem], i}
            prevMap[nums[i]] = i